### PR TITLE
(#1573) Generify `org.cactoos.experimental` package. 

### PR DESCRIPTION
**Pull request by rocket-3**
For #1573.

- Generified parameter types in constructors of `org.cactoos.experimental.Threads` class
- Removed `todo` comment

Actually one class and one constructor, which contract, in my opinion, could be `relaxed`.

### CODE REVIEW

**Comment by rocket-3** [comment]
@0crat status

**Comment by rocket-3** [comment]
@0crat, status

**Comment by 0crat** [comment]
Are you speaking to me or about me [here](https://github.com/yegor256/cactoos/issues/1627#issuecomment-921987795); you must always start your message with my name if you want to address it to me, see [§1](http://www.zerocracy.com/policy.html#1)

**Comment by victornoel** [comment]
@0crat status

**Comment by rocket-3** [comment]
@victornoel, how can I fix that failed code coverage check?

**Comment by victornoel** [comment]
@rocket-3 no need to, it's more of a glitch than an actual problem, minus 0.02% is the same as no changes. I never found how to prevent this unfortunately...

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1627#issuecomment-922332608))

@victornoel This is what I know about this job in C63314D6Z, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1627](https://github.com/yegor256/cactoos/issues/1627) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231627%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by victornoel** [comment]
@0crat in

**Comment by rocket-3** [comment]
@0crat status

File: src/main/java/org/cactoos/experimental/Threads.java [hunk 1]
```diff
@@ -188,7 +188,7 @@ public Threads(
      * @param tasks The tasks to be executed concurrently.
      */
     private Threads(
-        final Func<Iterable<Callable<T>>, Iterable<Future<T>>> fnc,
+        final Func<? super Iterable<Callable<T>>, ? extends Iterable<? extends Future<T>>> fnc,
```
Comment (victornoel) [inline src/main/java/org/cactoos/experimental/Threads.java:191]: @rocket-3 I think we are missing some in there, for the first `Iterable`, the `Callable` as well as the `Future`.

**Review by victornoel — CHANGES_REQUESTED** [review]

File: src/main/java/org/cactoos/experimental/Threads.java (hunk 1 above)
Comment (rocket-3) [reply src/main/java/org/cactoos/experimental/Threads.java]: @victornoel OK, as I see, it's a private ctor., therefore, it should be remained intact, as I think.
So I changed the PR to just puzzle removal commit.
WDYT?

Comment (victornoel) [reply src/main/java/org/cactoos/experimental/Threads.java]: @rocket-3 why not indeed

**Comment by victornoel** [comment]
@rultor merge

**Pull request merged**